modules/date/date-parser.c

```diff
@@ -148,7 +154,7 @@ date_parser_clone(LogPipe *s)
   LogParser *cloned;
 
   cloned = date_parser_new(log_pipe_get_config(&self->super.super));
-  date_parser_set_format(cloned, self->date_format);
+  date_parser_set_format(cloned, g_list_copy_deep(self->date_formats, (GCopyFunc) g_strdup, NULL));
   date_parser_set_timezone(cloned, self->date_tz);
```

Comment:
hopefully there's a string list helper that clones the list, if not one could be created.